### DISCUSSION TITLE
How can I prevent a RigidBody from passing through another RigidBody?

### DISCUSSION THREAD

**Discussion by arkk200** [pmndrs/react-three-rapier#474] [Q&A] [answered]
I am currently developing a feature where a dice rolls inside a cup when the cup is shaken or flipped. However, I am facing an issue where the dice sometimes passes through the sides of the cup instead of staying inside. How can I solve this problem?

**Comment by wiledal** [comment, marked as answer]
Hey! This can be a bit hard to achieve. Especially if the cup has an arbitrary shape (convexMesh), as it then does not have calculated "thickness". One thing you can try is to turn on `ccd` on the cup, or dice, or both.

Other people have reported that changing the setting for `predicitionDistance` to something like `0.09` (depending on the size of your scene), might help.

```tsx
<Physics predictionDistance={0.09}>...</Physics>
```
